src/main/java/com/ericsson/ei/services/SubscriptionService.java

```diff
@@ -56,9 +58,17 @@ public boolean addSubscription(Subscription subscription) {
     }
     
     @Override
-    public Subscription getSubscription(String name) throws SubscriptionNotFoundException {
+    public Subscription getSubscription(String name, String userName) throws SubscriptionNotFoundException {
         
         String query = String.format(SUBSCRIPTION_NAME, name);
+
+        if(!userName.isEmpty()){  
+            String queryUser = String.format(USER_NAME, userName);      
+            String operatorAnd = "{$and:[%s]}";
+            String queryTemp = query+ "," + queryUser;            
+            query = String.format(operatorAnd, queryTemp);            
+        }
```

Comment:
I don't understand why you need to query for both SUBSCRIPTION_NAME and USER_NAME in same query. It can only exist one 'SUBSCRIPTION_NAME' and adding USER_NAME to the query will make same result as query only SUBSCRIPTION_NAME.
Or have I miss-understood the functionality in this function?
After it have found the SUBSCRIPTION_NAME document,, then you can validate if the subscription document belongs to the current user.